src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -114,60 +144,110 @@ export const useBalanceTableColumns = (
+) => {
+  const { formatMessage } = useIntl();
+  const {
+    actionSidebarToggle: [, { toggleOn: toggleActionSidebarOn }],
+  } = useActionSidebarContext();
+
+  const getMenuProps = useCallback<
     TableWithMeatballMenuProps<BalanceTableFieldModel>['getMenuProps']
-  >(() => {
-    // @TODO: add actions and translations to every item
-    return {
-      cardClassName: 'min-w-[9.625rem] whitespace-nowrap',
-      items: [
-        {
-          key: 'add_funds',
-          onClick: () => {},
-          label: '',
-          icon: '',
-        },
-        // {
-        //   key: 'view_ethscan',
-        //   onClick: () => {},
-        //   icon: 'arrow-square-out',
-        //   label: formatText(
-        //     { id: 'membersPage.memberNav.viewOn' },
-        //     {
-        //       networkName: DEFAULT_NETWORK_INFO.blockExplorerName,
-        //     },
-        //   ),
-        //   renderItemWrapper: (props, children) => (
-        //     <Link
-        //       to={getBlockExplorerLink({
-        //         linkType: 'address',
-        //         addressOrHash: walletAddress,
-        //       })}
-        //       {...props}
-        //     >
-        //       {children}
-        //     </Link>
-        //   ),
-        // },
-        // {
-        //   key: 'mint_tokens',
-        //   onClick: () => {},
-        //   label: 'Mint tokens',
-        //   icon: 'bank',
-        // },
-        // {
-        //   key: 'transfer_fundss',
-        //   onClick: () => {},
-        //   label: 'Transfer funds',
-        //   icon: 'transfer',
-        // },
-        // {
-        //   key: 'make_payment',
-        //   onClick: () => {},
-        //   label: 'Make payment with token',
-        //   icon: 'hand-coins',
-        // },
-      ],
-    };
-  }, []);
+  >(
+    ({ index }) => {
+      const selectedTokenData = data[index]?.token;
+      const isTokenNative =
+        selectedTokenData?.tokenAddress === nativeToken?.tokenAddress;
+
+      return {
+        cardClassName: 'min-w-[9.625rem] whitespace-nowrap',
+        items: [
+          ...(!isTokenNative && !nativeTokenStatus?.unlocked
+            ? [
+                {
+                  key: 'add_funds',
+                  onClick: () => {
+                    toggleAddFundsModalOn();
+                  },
+                  label: formatMessage(MSG.labelAddFunds),
+                  icon: 'add',
+                },
+              ]
+            : []),
+          {
+            key: 'view_ethscan',
+            renderItemWrapper: (props, children) => (
+              <Link
+                to={getBlockExplorerLink({
+                  linkType: 'token',
+                  addressOrHash: selectedTokenData?.tokenAddress || '',
+                })}
+                {...props}
+              >
+                {children}
+              </Link>
+            ),
+            label: formatMessage(MSG.labelEthscan),
+            icon: 'arrow-square-out',
+          },
```

Comment:
This needs to use the correct name and link for appropriate network the CDapp is deployed to, exactly the same as the members page tooltip uses

https://github.com/JoinColony/colonyCDapp/blob/feat/15631447-balance-page-handling-table-actions/src/components/frame/v5/pages/MembersPage/partials/MembersTabContent/hooks.tsx#L64-L84